### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by perrettecl**

### CODE REVIEW

**Review by eric3873 — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,8 +30,18 @@ etn_target(interface ${PROJECT_NAME}
 )
 
 ##############################################################################################################
-
-if (BUILD_TESTING)
-    enable_testing()
-    add_subdirectory(test)
-endif()
+etn_test_target(${PROJECT_NAME}
```
Comment (etnKlendathu) [inline CMakeLists.txt:33]: Thank you for this :)

File: fty/process.h [hunk 2]
```diff
@@ -214,55 +216,71 @@ inline Expected<pid_t> Process::run()
...
-        }
+    // Wait with timeout is hard to do has the OS does not provid direct solution for it.
+    // We will do active waiting with step...
+    // This solution is thread safe
+    if (timeoutMs >= 0) {
 
-        timespec ts;
-        ts.tv_sec  = milliseconds / 1000;
-        ts.tv_nsec = (milliseconds % 1000) * 1000000;
+        if (waitCycleDurationMs == 0) {
```
Comment (etnKlendathu) [inline fty/process.h:229]: could you add assert here? would be nice in debug mode

File: fty/process.h [hunk 3]
```diff
@@ -214,55 +216,71 @@ inline Expected<pid_t> Process::run()
...
+                    return WSTOPSIG(status);
+                }
             }
         }
-        m_pid = 0;
-        return status;
     }
 
+    // We wait until the end of the program -> wait pid can return for several reason, and we want to get only when our pid returns
     do {
```
Comment (eric3873) [inline fty/process.h:279]: Dead code ?

File: fty/process.h [hunk 4]
```diff
@@ -214,55 +216,71 @@ inline Expected<pid_t> Process::run()
...
+                if (WIFEXITED(status)) {
+                    m_pid = 0;
+                    return WEXITSTATUS(status);
+                } else if (WIFSIGNALED(status)) {
+                    m_pid = 0;
+                    return WTERMSIG(status);
+                } else if (WIFSTOPPED(status)) {
+                    m_pid = 0;
+                    return WSTOPSIG(status);
+                }
```
[the remark this answers is not in the document]
Comment (perrettecl) [reply fty/process.h:273]: Good catch!

**Pull request merged**